### PR TITLE
Add `dfu-programmer` to `pacman -S` (#6618)

### PR DESCRIPTION
**Pull request by Cory-Watson**
`dfu-programmer` now resides at `extra/dfu-programmer` and is no longer
in the AUR

* Add `--needed` option to `pacman -S` for efficiency

* Fix

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* #6618 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: util/linux_install.sh [hunk 1]
```diff
@@ -68,6 +68,7 @@ elif grep ID /etc/os-release | grep -q 'arch\|manjaro'; then
 		avr-libc \
 		avr-gcc \
 		base-devel \
+        dfu-programmer \
```
Comment (fauxpark) [inline util/linux_install.sh]: Just a minor thing, but this line should be indented with tabs instead of spaces, to match the rest of the file.

**Comment by yanfali** [comment]
@Cory-Watson we use editor-config, there's a plugin for vim, that should prevent your defaults from conflicting with QMKs.

**Comment by Cory-Watson** [comment]
@yanfali thanks for the tip. editor-config seems to be really awesome!

**Comment by drashna** [comment]
Also, to update a PR, you just need to add commits to the original branch, and it will update the PR.

**Comment by drashna** [comment]
Looks like there is a merge conflict here.

Could you resolve that?

**Comment by Cory-Watson** [comment]
Should I add
```
[linux_install.sh]
indent_style = tab
```
to `.editorconfig`?

**Comment by drashna** [comment]
Maybe, but I think that would be outside the scope of this PR.

**Pull request merged**